src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Controls/DataGridCell.cs

```diff
@@ -964,6 +964,28 @@ protected override void OnPreviewKeyDown(KeyEventArgs e)
         /// </summary>
         protected override void OnKeyDown(KeyEventArgs e)
         {
+            if ((Keyboard.Modifiers & ModifierKeys.Alt) == ModifierKeys.Alt)
+            {
+                DataGridLength updatedWidth = new DataGridLength();
+
+                if (e.Key == Key.Right)
+                {
+                    updatedWidth = new DataGridLength(this.Column.ActualWidth + ColumnWidthStepSize);
+                }
+                else if (e.Key == Key.Left)
+                {
+                    updatedWidth = new DataGridLength(this.Column.ActualWidth - ColumnWidthStepSize);
+                }
+
+                if (Column.CanColumnResize(updatedWidth))
+                {
+                    this.Column.SetCurrentValue(DataGridColumn.WidthProperty, updatedWidth);
+                }
+
+                e.Handled = true;
```

Comment:
You should only do these lines when e.Key is Right or Left.  As it stands, typing Alt+Up (for example) results in setting column width to 0 (default DataGridLength) and handling the event, both of which are wrong.